sktime/forecasting/es_rnn.py

```diff
@@ -0,0 +1,309 @@
+        self.pred_len = pred_len
+        self.stride = stride
+        self._get_data()
+
+    def _get_data(self):
+        length = len(self.y)
+        x_arr = []
+        y_arr = []
+        for i in range(0, length - self.window - self.pred_len + 1, self.stride):
+            inp = self.y[i : i + self.window]
+            out = self.y[i + self.window : i + self.window + self.pred_len]
+
+            x_arr.append(inp)
+            y_arr.append(out)
+
+        if not x_arr:
+            raise ValueError("Input size to small")
+
+        self.x_train, self.y_train = (
+            torch.FloatTensor(np.array(x_arr)),
+            torch.FloatTensor(np.array(y_arr)),
+        )
+
+    def __len__(self):
+        """Get length of the dataset."""
+        return len(self.x_train)
+
+    def __getitem__(self, idx):
+        """Get data pairs at this index."""
+        return self.x_train[idx], self.y_train[idx]
+
+
+class ESRNNPredDataset(Dataset):
+    """Implements Pytorch Dataset class for Prediction ESS-RNN."""
+
+    def __init__(self, y, window) -> None:
+        self.y = y
+        self.window = window
+        self._get_data()
+
+    def _get_data(self):
+        x_pred = self.y[-self.window :]
+        x_pred = torch.FloatTensor(np.array(x_pred))
+        self.x_pred = x_pred.unsqueeze(0)
+
+    def __len__(self):
+        """Get length of the dataset."""
+        return len(self.x_pred)
+
+    def __getitem__(self, idx):
+        """Return data point."""
+        return self.x_pred[idx], torch.zeros(1)
+
+
+class ESRNNForecaster(BaseDeepNetworkPyTorch):
+    """
+    Exponential Smoothing Recurrant Neural Network.
+
+    This model combines Exponential Smoothing (ES) and (LSTM) networks
+    for time series forecasting. ES is used to balance the level and
+    seasonality of the series. This method has been proposed in [1]_.
+
+    Parameters
+    ----------
+    input_shape : int
+        Number of features in the input
+    hidden_size : int
+        Number of features in the hidden state
+    num_layer : int
+        Number of layers
+    seasonality_type : string
+        Type of seasonality_type, could be zero ,single or double
+    season1_length : int
+        Period of season 1
+    season2_length : int
+        Period of season 2
+    stride : int
+        stride for sliding window
+    batch_size : int
+        size of batch during training
+    num_epochs : int
+        number of epochs during training
+    criterion : torch.nn Loss Function, default=torch.nn.MSELoss
+        loss function to be used for training
+    criterion_kwargs : dict, default=None
+        keyword arguments to pass to criterion
+    optimizer : torch.optim.Optimizer, default=torch.optim.Adam
+        optimizer to be used for training
+    optimizer_kwargs : dict, default=None
+        keyword arguments to pass to optimizer
+    window : int
+        Size of Input window, default=5
+    lr_rate : int
+        Learning rate for training
```

Comment:
```suggestion
    lr : int
        Learning rate for training
```
should be simply called `lr` (learning rate) instead, fix at all places